docs/workshop/index.md

```diff
@@ -0,0 +1,410 @@
+> A service is used to collect information from the user and the map then
+use that information to generate a query.  After the query has executed, the results are then rendered
+into HTML.
+
+* Not all `<map-source>`es are equal!
+Previous versions of GeoMoose used PHP to query data.  Now, GeoMoose uses WMS, WFS,
+and ArcGIS FeatureServices to perform those queries. To get full querying capabilities
+from a layer it needs to have a `mapserver-wfs`, `wfs` or `ags-vector` source.
+
+## Adding select
+
+* To be able to select features from a layer it needs to be configured as WFS.
+* The following needs added after line 5 in `mapbook.xml`:
+
+```xml
+<map-source name="firestations-wfs" type="mapserver-wfs">
+    <file>./demo/firestations/firestations.map</file>
+    <param name="typename" value="ms:fire_stations" />
+    <layer name="fire_stations" selectable="true" title="Firestations">
+        <template name="select"><![CDATA[
+        <div class="result-item">
+            <div class="result-title">
+            Firestation
+            </div>
+            <b>Station City:</b> {{ properties.Dak_GIS__4 }}<br>
+            <b>Station Number:</b> {{ properties.Dak_GIS__5 }}<br>
+        <div>
+        ]]></template>
+    </layer>
+</map-source>
+```
+
+## Reload and select Firestations
+
+* Back in the browser, reload the page, and click 'Select' from the toolbar.
+* Use the dropdown to select "Firestations", draw a Polygon on the map, and click "Go".
+
+![Selecting firestations](./images/firestations-select.png)
+
+## More on vector layers
+
+* GeoMoose 3 has been designed with a "vector first" philosophy.
+* MapBox GL Styles can be used to style vector layers for rendering in the browser.
+* There are additional examples in the example `mapbook.xml`.
+
+# Customizing the GeoMoose application
+
+## Configuring GeoMoose components
+
+* Open `app.js` in an editor.
+* The gm3.Application's `add` function accepts three paramters:
+  1. The component class.
+  2. The DOM element ID where the component.
+  3. The optinal properties for the component.
+
+## Configure Coordinate Display
+
+* The coordinate display shows the mouse position as the user moves
+  the cursor across the map.
+* This X,Y is in the map's configured coordinate system. As the
+  examples are configured to use the Google Maps projection, that is
+  not going to be useful.
+* In `app.js` on line 60, you can see the definition of the UTM-15 projection:
+    ```
+    app.addProjection({
+        ref: 'EPSG:26915',
+        def: '+proj=utm +zone=15 +ellps=GRS80 +datum=NAD83 +units=m +no_defs'
+    });
+    ```
+* To use UTM-15 make the following change:
+    ```
+            ref: 'xy'
+    ```
+    to:
+    ```
+            ref: 'EPSG:26915'
+    ```
+
+
+## GeoMoose is very HTML, such CSS.
+
+* Chrome and Firefox both have excellent DOM/CSS Inspection tools.
+* The GeoMoose example `index.html` includes a file it won't find `site.css`.
```

Comment:
Reads really weird in PDF.

find<add , or : or something> `site.css`... or move `site.css` after file separated by commas.